From 08f47166ed0b547ebff58665a2015a1ca8a6bcf0 Mon Sep 17 00:00:00 2001
From: Huaxiang Sun <hsun@cloudera.com>
Date: Wed, 2 Dec 2015 16:27:47 -0800
Subject: [PATCH 079/197] CLOUDERA-BUILD Change build to point to Cloudera versions and repositories.

Changes in hbase-resource-bundle/src/main/resources/supplemental-models.xml
needs revisit for proper license

Change-Id: Icfd1cb16c7f76bb4dcfcd7e7753dc756cbfe14b8
---
 hbase-annotations/pom.xml                          |    2 +-
 hbase-assembly/pom.xml                             |    4 +-
 hbase-checkstyle/pom.xml                           |    4 +-
 hbase-client/pom.xml                               |    4 +-
 hbase-common/pom.xml                               |    2 +-
 hbase-examples/pom.xml                             |    2 +-
 hbase-external-blockcache/pom.xml                  |    2 +-
 hbase-hadoop-compat/pom.xml                        |    2 +-
 hbase-hadoop2-compat/pom.xml                       |    2 +-
 hbase-it/pom.xml                                   |    4 +-
 hbase-prefix-tree/pom.xml                          |    4 +-
 hbase-procedure/pom.xml                            |    2 +-
 hbase-protocol/pom.xml                             |    2 +-
 hbase-resource-bundle/pom.xml                      |    2 +-
 .../src/main/resources/supplemental-models.xml     |   14 +++
 hbase-rest/pom.xml                                 |    4 +-
 hbase-server/pom.xml                               |    6 +-
 hbase-shaded/hbase-shaded-client/pom.xml           |    2 +-
 hbase-shaded/hbase-shaded-server/pom.xml           |    2 +-
 hbase-shaded/pom.xml                               |    2 +-
 hbase-shell/pom.xml                                |    4 +-
 hbase-testing-util/pom.xml                         |    6 +-
 hbase-thrift/pom.xml                               |    4 +-
 pom.xml                                            |   93 ++++++++++++++++----
 24 files changed, 122 insertions(+), 53 deletions(-)

diff --git a/hbase-annotations/pom.xml b/hbase-annotations/pom.xml
index 776710c..6d47680 100644
--- a/hbase-annotations/pom.xml
+++ b/hbase-annotations/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
diff --git a/hbase-assembly/pom.xml b/hbase-assembly/pom.xml
index 750b07e..a633818 100644
--- a/hbase-assembly/pom.xml
+++ b/hbase-assembly/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-assembly</artifactId>
@@ -167,7 +167,7 @@
     </dependency>
     <dependency>
         <groupId>org.apache.hbase</groupId>
-        <artifactId>${compat.module}</artifactId>
+        <artifactId>hbase-hadoop2-compat</artifactId>
         <version>${project.version}</version>
     </dependency>
     <dependency>
diff --git a/hbase-checkstyle/pom.xml b/hbase-checkstyle/pom.xml
index 52c8e56..f825b63 100644
--- a/hbase-checkstyle/pom.xml
+++ b/hbase-checkstyle/pom.xml
@@ -24,14 +24,14 @@
 <modelVersion>4.0.0</modelVersion>
 <groupId>org.apache.hbase</groupId>
 <artifactId>hbase-checkstyle</artifactId>
-<version>1.2.0</version>
+<version>1.2.0-cdh5.7.0-SNAPSHOT</version>
 <name>Apache HBase - Checkstyle</name>
 <description>Module to hold Checkstyle properties for HBase.</description>
 
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
diff --git a/hbase-client/pom.xml b/hbase-client/pom.xml
index 57f4f94..0a5d09d 100644
--- a/hbase-client/pom.xml
+++ b/hbase-client/pom.xml
@@ -24,7 +24,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
@@ -285,7 +285,7 @@
             <exclusion>
               <groupId>com.sun.jersey</groupId>
               <artifactId>jersey-json</artifactId>
-            </exclusion> 
+            </exclusion>
             <exclusion>
               <groupId>javax.servlet</groupId>
               <artifactId>servlet-api</artifactId>
diff --git a/hbase-common/pom.xml b/hbase-common/pom.xml
index a115657..360b9c9 100644
--- a/hbase-common/pom.xml
+++ b/hbase-common/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
diff --git a/hbase-examples/pom.xml b/hbase-examples/pom.xml
index 4272286..375ca00 100644
--- a/hbase-examples/pom.xml
+++ b/hbase-examples/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-examples</artifactId>
diff --git a/hbase-external-blockcache/pom.xml b/hbase-external-blockcache/pom.xml
index 1a0bf40..e83db77 100644
--- a/hbase-external-blockcache/pom.xml
+++ b/hbase-external-blockcache/pom.xml
@@ -25,7 +25,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-external-blockcache</artifactId>
diff --git a/hbase-hadoop-compat/pom.xml b/hbase-hadoop-compat/pom.xml
index 3107024..540c7d6 100644
--- a/hbase-hadoop-compat/pom.xml
+++ b/hbase-hadoop-compat/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <artifactId>hbase</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>1.2.0</version>
+        <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
 
diff --git a/hbase-hadoop2-compat/pom.xml b/hbase-hadoop2-compat/pom.xml
index 39f7661..7f12f29 100644
--- a/hbase-hadoop2-compat/pom.xml
+++ b/hbase-hadoop2-compat/pom.xml
@@ -21,7 +21,7 @@ limitations under the License.
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
diff --git a/hbase-it/pom.xml b/hbase-it/pom.xml
index d7e6610..6bb0bfc 100644
--- a/hbase-it/pom.xml
+++ b/hbase-it/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
@@ -215,7 +215,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
diff --git a/hbase-prefix-tree/pom.xml b/hbase-prefix-tree/pom.xml
index 09c0c43..93b0b12 100644
--- a/hbase-prefix-tree/pom.xml
+++ b/hbase-prefix-tree/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
@@ -126,7 +126,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
diff --git a/hbase-procedure/pom.xml b/hbase-procedure/pom.xml
index be4bb09..3046373 100644
--- a/hbase-procedure/pom.xml
+++ b/hbase-procedure/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
diff --git a/hbase-protocol/pom.xml b/hbase-protocol/pom.xml
index 16d2964..c4d87c4 100644
--- a/hbase-protocol/pom.xml
+++ b/hbase-protocol/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <artifactId>hbase</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>1.2.0</version>
+        <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
 
diff --git a/hbase-resource-bundle/pom.xml b/hbase-resource-bundle/pom.xml
index ebbbe36..6ee422a 100644
--- a/hbase-resource-bundle/pom.xml
+++ b/hbase-resource-bundle/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
diff --git a/hbase-resource-bundle/src/main/resources/supplemental-models.xml b/hbase-resource-bundle/src/main/resources/supplemental-models.xml
index d01f188..7acc962 100644
--- a/hbase-resource-bundle/src/main/resources/supplemental-models.xml
+++ b/hbase-resource-bundle/src/main/resources/supplemental-models.xml
@@ -1195,4 +1195,18 @@ Copyright (c) 2007-2011 The JRuby project
       </licenses>
     </project>
   </supplement>
+  <supplement>
+    <project>
+      <groupId>javax.servlet.jsp</groupId>
+      <artifactId>jsp-api</artifactId>
+
+      <licenses>
+        <license>
+          <name>CDDL + GPLv2 with classpath exception</name>
+          <url>https://glassfish.dev.java.net/nonav/public/CDDL+GPL.html</url>
+          <distribution>repo</distribution>
+        </license>
+      </licenses>
+    </project>
+  </supplement>
 </supplementalDataModels>
diff --git a/hbase-rest/pom.xml b/hbase-rest/pom.xml
index d91db79..36c1b8c 100644
--- a/hbase-rest/pom.xml
+++ b/hbase-rest/pom.xml
@@ -25,7 +25,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-rest</artifactId>
@@ -210,7 +210,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
diff --git a/hbase-server/pom.xml b/hbase-server/pom.xml
index 45da7dc..f18fa92 100644
--- a/hbase-server/pom.xml
+++ b/hbase-server/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-server</artifactId>
@@ -428,12 +428,12 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
       <type>test-jar</type>
       <scope>test</scope>
diff --git a/hbase-shaded/hbase-shaded-client/pom.xml b/hbase-shaded/hbase-shaded-client/pom.xml
index 12fe62d..a49c957 100644
--- a/hbase-shaded/hbase-shaded-client/pom.xml
+++ b/hbase-shaded/hbase-shaded-client/pom.xml
@@ -24,7 +24,7 @@
     <parent>
         <artifactId>hbase-shaded</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>1.2.0</version>
+        <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
     <artifactId>hbase-shaded-client</artifactId>
diff --git a/hbase-shaded/hbase-shaded-server/pom.xml b/hbase-shaded/hbase-shaded-server/pom.xml
index 9b3083f..8989a4c 100644
--- a/hbase-shaded/hbase-shaded-server/pom.xml
+++ b/hbase-shaded/hbase-shaded-server/pom.xml
@@ -24,7 +24,7 @@
     <parent>
         <artifactId>hbase-shaded</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>1.2.0</version>
+        <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
     <artifactId>hbase-shaded-server</artifactId>
diff --git a/hbase-shaded/pom.xml b/hbase-shaded/pom.xml
index 486e4f0..0e3cf76 100644
--- a/hbase-shaded/pom.xml
+++ b/hbase-shaded/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <artifactId>hbase</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>1.2.0</version>
+        <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
     <artifactId>hbase-shaded</artifactId>
diff --git a/hbase-shell/pom.xml b/hbase-shell/pom.xml
index 921b2b7..875c796 100644
--- a/hbase-shell/pom.xml
+++ b/hbase-shell/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-shell</artifactId>
@@ -226,7 +226,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
diff --git a/hbase-testing-util/pom.xml b/hbase-testing-util/pom.xml
index 01de6a4..d942dd9 100644
--- a/hbase-testing-util/pom.xml
+++ b/hbase-testing-util/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <artifactId>hbase</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>1.2.0</version>
+        <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
     <artifactId>hbase-testing-util</artifactId>
@@ -104,13 +104,13 @@
         </dependency>
         <dependency>
             <groupId>org.apache.hbase</groupId>
-            <artifactId>${compat.module}</artifactId>
+            <artifactId>hbase-hadoop2-compat</artifactId>
             <type>jar</type>
             <scope>compile</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.hbase</groupId>
-            <artifactId>${compat.module}</artifactId>
+            <artifactId>hbase-hadoop2-compat</artifactId>
             <type>test-jar</type>
             <scope>compile</scope>
         </dependency>
diff --git a/hbase-thrift/pom.xml b/hbase-thrift/pom.xml
index 0d923ab..7fb883f 100644
--- a/hbase-thrift/pom.xml
+++ b/hbase-thrift/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>1.2.0</version>
+    <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-thrift</artifactId>
@@ -281,7 +281,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
diff --git a/pom.xml b/pom.xml
index d04c831..8cb5e28 100644
--- a/pom.xml
+++ b/pom.xml
@@ -30,16 +30,16 @@
 -->
   <modelVersion>4.0.0</modelVersion>
   <parent>
-    <groupId>org.apache</groupId>
-    <artifactId>apache</artifactId>
-    <version>12</version>
+    <groupId>com.cloudera.cdh</groupId>
+    <artifactId>cdh-root</artifactId>
+    <version>5.7.0-SNAPSHOT</version>
     <relativePath/>
     <!-- no parent resolution -->
   </parent>
   <groupId>org.apache.hbase</groupId>
   <artifactId>hbase</artifactId>
   <packaging>pom</packaging>
-  <version>1.2.0</version>
+  <version>1.2.0-cdh5.7.0-SNAPSHOT</version>
   <name>Apache HBase</name>
   <description>
     Apache HBase is the Hadoop database. Use it when you need
@@ -417,6 +417,29 @@
   </developers>
   <repositories>
     <repository>
+      <id>cdh.repo</id>
+      <url>https://repository.cloudera.com/artifactory/cloudera-repos</url>
+      <name>Cloudera Repositories</name>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </repository>
+    <repository>
+      <id>cdh.snapshots.repo</id>
+      <url>https://repository.cloudera.com/artifactory/libs-snapshot-local</url>
+      <name>Cloudera Snapshots Repository</name>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+      <releases>
+        <enabled>false</enabled>
+      </releases>
+    </repository>
+    <repository>
+      <id>cloudbees netty</id>
+      <url>http://repository-netty.forge.cloudbees.com/snapshot/</url>
+    </repository>
+    <repository>
       <id>apache release</id>
       <url>https://repository.apache.org/content/repositories/releases/</url>
     </repository>
@@ -524,13 +547,19 @@
             <systemPropertyVariables>
               <test.build.classes>${test.build.classes}</test.build.classes>
             </systemPropertyVariables>
-
+            <includes>
+              <include>${unittest.include}</include>
+            </includes>
             <excludes>
-	      <!-- users can add -D option to skip particular test classes
-		   ex: mvn test -Dtest.exclude.pattern=**/TestFoo.java,**/TestBar.java
-	      -->
-	      <exclude>${test.exclude.pattern}</exclude>
-	    </excludes>
+              <exclude>${integrationtest.include}</exclude>
+              <exclude>**/*$*</exclude>
+              <exclude>${test.exclude.pattern}</exclude>
+            </excludes>
+            <environmentVariables>
+              <LD_LIBRARY_PATH>${env.LD_LIBRARY_PATH}:${project.build.directory}/nativelib</LD_LIBRARY_PATH>
+              <DYLD_LIBRARY_PATH>${env.DYLD_LIBRARY_PATH}:${project.build.directory}/nativelib</DYLD_LIBRARY_PATH>
+              <MALLOC_ARENA_MAX>4</MALLOC_ARENA_MAX>
+            </environmentVariables>
           </configuration>
           <executions>
             <execution>
@@ -766,6 +795,8 @@
               <exclude>src/main/site/resources/repo/**</exclude>
               <exclude>**/dependency-reduced-pom.xml</exclude>
               <exclude>**/rat.txt</exclude>
+              <exclude>**/cloudera/**</exclude>
+              <exclude>**/debian/**</exclude>
             </excludes>
           </configuration>
         </plugin>
@@ -1137,8 +1168,8 @@
     <maven.min.version>3.0.3</maven.min.version>
     <java.min.version>${compileSource}</java.min.version>
     <!-- Dependencies -->
-    <hadoop-two.version>2.5.1</hadoop-two.version>
     <!-- minikdc introduced in hadoop-2.3; override when building earlier versions -->
+    <hadoop-two.version>${cdh.hadoop.version}</hadoop-two.version>
     <hadoop-two-minikdc.version>${hadoop-two.version}</hadoop-two-minikdc.version>
     <hadoop-three.version>3.0.0-SNAPSHOT</hadoop-three.version>
     <commons-cli.version>1.2</commons-cli.version>
@@ -1156,10 +1187,10 @@
     <metrics-core.version>2.2.0</metrics-core.version>
     <guava.version>12.0.1</guava.version>
     <jsr305.version>1.3.9</jsr305.version>
-    <jackson.version>1.9.13</jackson.version>
+    <jackson.version>${cdh.jackson.version}</jackson.version>
     <jasper.version>5.5.23</jasper.version>
-    <jaxb-api.version>2.2.2</jaxb-api.version>
-    <jetty.version>6.1.26</jetty.version>
+    <jaxb-api.version>2.1</jaxb-api.version>
+    <jetty.version>6.1.26.cloudera.4</jetty.version>
     <jetty.jspapi.version>6.1.14</jetty.jspapi.version>
     <jersey.version>1.9</jersey.version>
     <jruby.version>1.6.8</jruby.version>
@@ -1171,9 +1202,11 @@
     <protobuf.version>2.5.0</protobuf.version>
     <thrift.path>thrift</thrift.path>
     <thrift.version>0.9.3</thrift.version>
-    <zookeeper.version>3.4.6</zookeeper.version>
-    <slf4j.version>1.7.7</slf4j.version>
     <clover.version>4.0.3</clover.version>
+    <zookeeper.version>${cdh.zookeeper.version}</zookeeper.version>
+    <slf4j.version>${cdh.slf4j.version}</slf4j.version>
+    <hadoop-snappy.version>${cdh.hadoop-snappy.version}</hadoop-snappy.version>
+    <jets3.version>0.9.0</jets3.version>
     <jamon-runtime.version>2.4.1</jamon-runtime.version>
     <jettison.version>1.3.3</jettison.version>
     <netty.version>4.0.23.Final</netty.version>
@@ -1216,7 +1249,12 @@
     <surefire.testFailureIgnore>false</surefire.testFailureIgnore>
     <test.output.tofile>true</test.output.tofile>
     <surefire.timeout>900</surefire.timeout>
-    <test.exclude.pattern></test.exclude.pattern>
+    <!-- For flaky tests exclusion -->
+    <test.exclude></test.exclude>
+    <test.exclude.pattern>**/${test.exclude}.java</test.exclude.pattern>
+    <!-- Test inclusion patterns -->
+    <unittest.include>**/Test*.java</unittest.include>
+    <integrationtest.include>**/IntegrationTest*.java</integrationtest.include>
     <!-- default Xmx value is 2800m. Use -Dsurefire.Xmx=xxg to run tests with different JVM Xmx value -->
     <surefire.Xmx>2800m</surefire.Xmx>
     <surefire.cygwinXmx>2800m</surefire.cygwinXmx>
@@ -1300,12 +1338,12 @@
       </dependency>
       <dependency>
         <groupId>org.apache.hbase</groupId>
-        <artifactId>${compat.module}</artifactId>
+        <artifactId>hbase-hadoop2-compat</artifactId>
         <version>${project.version}</version>
       </dependency>
       <dependency>
         <groupId>org.apache.hbase</groupId>
-        <artifactId>${compat.module}</artifactId>
+        <artifactId>hbase-hadoop2-compat</artifactId>
         <version>${project.version}</version>
         <type>test-jar</type>
         <scope>test</scope>
@@ -1719,6 +1757,23 @@
        <version>${bouncycastle.version}</version>
        <scope>test</scope>
      </dependency>
+     <dependency>
+       <groupId>net.java.dev.jets3t</groupId>
+       <artifactId>jets3t</artifactId>
+       <version>${jets3.version}</version>
+     </dependency>
+     <dependency>
+       <groupId>org.jmock</groupId>
+       <artifactId>jmock-junit4</artifactId>
+       <version>${jmock-junit4.version}</version>
+       <scope>test</scope>
+       <exclusions>
+         <exclusion>
+           <artifactId>junit-dep</artifactId>
+           <groupId>junit</groupId>
+         </exclusion>
+       </exclusions>
+     </dependency>
     </dependencies>
   </dependencyManagement>
   <!-- Dependencies needed by subprojects -->
-- 
1.7.0.4

